app/models/task.py

```diff
@@ -1,5 +1,34 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from sqlalchemy import ForeignKey
 from ..db import db
+from datetime import datetime
+from typing import Optional
+
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    description: Mapped[str]
+    completed_at: Mapped[Optional[datetime]]
+    goal_id: Mapped[Optional[int]] = db.Column(db.Integer, db.ForeignKey("goal.id"))
```

Comment:
You're mixing syntax in this model definition. In some places you use the declarative mapping style, which is the standard now and what we teach in the Learn curriculum. 

Lines 9-12 for example uses declarative mapping. Then here you use the classical style (also called imperative style) of mapping which is no longer recommended. Be sure to double check that your code follows the most-up-to-date syntax/convention and that you are being consistent. 

Here on line 13, I'd expect this to look like:
```suggestion
    goal_id: Mapped[Optional[int]] = mapped_column(ForeignKey("goal.id"))
```

You can review the documentation about the mapping styles [here](https://docs.sqlalchemy.org/en/20/orm/mapping_styles.html)